exchanges/gateio/gateio.go

```diff
@@ -424,169 +949,2870 @@ func (g *Gateio) SendAuthenticatedHTTPRequest(ctx context.Context, ep exchange.U
 		return err
 	}
 	headers := make(map[string]string)
-	headers["Content-Type"] = "application/x-www-form-urlencoded"
-	headers["key"] = creds.Key
-
-	hmac, err := g.GenerateSignature(creds.Secret, param)
+	urlPath := endpoint
+	timestamp := time.Now()
+	var paramValue string
+	if param != nil {
+		paramValue = param.Encode()
+	}
+	hmac, err := g.GenerateSignature(creds.Secret, method, "/"+gateioAPIVersion+endpoint, paramValue, data, timestamp)
```

Comment:
```suggestion
	sig, err := g.GenerateSignature(creds.Secret, method, "/"+gateioAPIVersion+endpoint, paramValue, data, timestamp)
```
As `hmac` conflicts with package name